gemfire-core/src/main/java/com/gemstone/gemfire/internal/cache/DiskEntry.java

```diff
@@ -614,37 +652,39 @@ else if (value == Token.TOMBSTONE) {
                 bytes = ohproxy.getSerializedValue();
               } else {
                 //TODO:Asif: Speak to Darrel for cleaner way
-                if(ohproxy instanceof ByteSource) {
+                if (ohproxy instanceof ByteSource) {
                   bytes = ((ByteSource)ohproxy).getRowBytes();
-                }else {
-                  bytes = (byte[]) ohproxy.getDeserializedForReading();
+                } else {
+                  bytes = (byte[])ohproxy.getDeserializedForReading();
                 }
               }
             } else {
               bytes = proxy.getSerializedValue();
             }
+            buffer = Helper.wrapBytes(bytes);
           }
           else if (value instanceof byte[]) {
             isSerializedObject = false;
-            bytes = (byte[])value;
+            buffer = ByteBuffer.wrap((byte[])value);
           }
           else {
             Assert.assertTrue(!Token.isRemovedFromDisk(value));
-            bytes = EntryEventImpl.serialize(value);
-            if (bytes.length == 0) {
-              throw new IllegalStateException("serializing <" + value + "> produced empty byte array");
+            buffer = EntryEventImpl.serializeDirect(value, null);
```

Comment:
I may be missing something but isn't this value going to be DirectByteBuffer for column data? If yes, this could be optimized.